test/unit/MemoryOptionsTest.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LaminasTest\Cache\Storage\Adapter;
+
+use Laminas\Cache;
+use Laminas\Cache\Exception\InvalidArgumentException;
+
+use function ini_set;
+
+final class MemoryOptionsTest extends AbstractCommonAdapterTest
```

Comment:
This is testing the options, IMO no need to test the whole adapter.
You can use `AbstractAdapterOptionsTest` instead.

Reply:
I changed it. But the alraedy existing MemoryTest is failing with the same error:

```
1) LaminasTest\Cache\Storage\Adapter\MemoryTest::testOptionsFluentInterface
Laminas\Cache\Exception\InvalidArgumentException: Memory limit must be an integer or string
```

Reply:
Can you point me to the test which tests `null` to be a valid value? 
That sounds kinda weird to me and I wonder how we are ending up in that error.
IMO, I'd disallow `null` at all but maybe we have something wrong in our test which leads to that problem.

Reply:
It is this already existing test:

```
1) LaminasTest\Cache\Storage\Adapter\MemoryTest::testOptionsFluentInterface
Laminas\Cache\Exception\InvalidArgumentException: Memory limit must be an integer or string
```

Looks like this test is just setting the current value of the options property and expects the same value back again. As there is nothing set for `memoryLimit`by default, the test throws an exception now instead of returning `null` which breaks the test.

Reply:
It is fixed now due to the default value for the `memoryLimit`, no?

Reply:
Yes, that is exactly the case. Setting the default is letting the test pass.